### ISSUE TITLE
Raise from previous error should not be triggered when the raised error is the caught error. 

### ISSUE THREAD

**Issue by Luttik** [sourcery-ai/sourcery#202]
### Issue description or question

Currently a codeblock like the one below will give the `raise-from-previous-error` advice even though it clearly should not happen:

```
except HttpError as e:
    if "already exists" not in e.reason:
        raise e
```

My advice is to check if the raised exception is the caught exception and if so not trigger the `raise-from-previous-error` rule. 

### Sourcery Version

Pycharm plugin 0.10.3

**Comment by ruancomelli** [comment]
Hi, @Luttik, thank you for opening this issue. You're totally right, it doesn't make much sense for this suggestion to trigger in this case. I was able to reproduce this and will work on a fix as soon as possible.

**Comment by Hellebore** [comment]
Fixed with release of 0.11.1
